inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/selfmonitoring/SelfMonitoringService.java

```diff
@@ -1,63 +1,62 @@
+import io.opencensus.stats.StatsRecorder;
 import io.opencensus.tags.TagKey;
 import io.opencensus.tags.TagValue;
 import io.opencensus.tags.Tags;
 import lombok.Data;
 import lombok.extern.slf4j.Slf4j;
+import lombok.val;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.event.EventListener;
 import org.springframework.stereotype.Component;
-import rocks.inspectit.oce.core.config.model.InspectitConfig;
-import rocks.inspectit.oce.core.service.DynamicallyActivatableService;
+import rocks.inspectit.oce.core.config.InspectitConfigChangedEvent;
+import rocks.inspectit.oce.core.config.InspectitEnvironment;
+import rocks.inspectit.oce.core.config.model.selfmonitoring.SelfMonitoringSettings;
+import rocks.inspectit.oce.core.metrics.MeasuresAndViewsProvider;
+import rocks.inspectit.oce.core.tags.CommonTagsManager;
 
-import java.util.Collections;
 import java.util.concurrent.TimeUnit;
-import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
 
 @Component
 @Slf4j
-public class SelfMonitoringService extends DynamicallyActivatableService {
+public class SelfMonitoringService {
+
+    private static final String DURATION_MEASURE_NAME = "duration";
+    private static final String DURATION_MEASURE_DESCRIPTION = "inspectIT OCE self-monitoring duration";
+    private static final String DURATION_MEASURE_UNIT = "μs";
 
     private static final TagKey COMPONENT_TAG_KEY = TagKey.create("component-name");
 
-    private static final AtomicReference<Measure.MeasureDouble> MEASURE_ATOMIC_REFERENCE = new AtomicReference<>();
 
     @Autowired
-    private StatsRecorder statsRecorder;
+    private InspectitEnvironment env;
 
     @Autowired
-    private ViewManager viewManager;
+    private StatsRecorder statsRecorder;
 
-    /**
-     * Measure name.
-     */
-    private String measureName;
+    @Autowired
+    private MeasuresAndViewsProvider measureProvider;
 
-    /**
-     * Created measure.
-     */
-    private Measure.MeasureDouble measure;
+    @Autowired
+    private CommonTagsManager commonTags;
 
-    /**
-     * Default constructor.
-     */
-    public SelfMonitoringService() {
-        super("selfMonitoring");
-    }
 
     /**
      * Provides an auto-closable that can be used in try-with-resource form.
      * <p>
      * If self monitoring is enabled the {@link SelfMonitoringScope} instance is create that handles time measuring and measure recording.
-     * If self monitoring is disabled, return no-ops closable.
+     * If self monitoring is disabled, returns a no-ops closable.
      *
      * @param componentName
      * @return
      */
-    public Scope withSelfMonitoring(String componentName) {
-        if (super.isEnabled()) {
+    public Scope withDurationSelfMonitoring(String componentName) {
+        if (isSelfMonitoringEnabled()) {
             return new SelfMonitoringScope(componentName, System.nanoTime());
         } else {
             return () -> {
```

Comment:
We always create a new instance here. Let's create one instance which we return every time.